### ISSUE TITLE
Some bug with key/resolvable/sharable/

### ISSUE THREAD

**Issue by flymedllva** [wundergraph/cosmo#824]
### Component(s)

router

### Component version

0.89.1

### wgc version

0.53.2

### controlplane version

-

### router version

0.89.1

## Description

I have different combination of key/resolvable/sharable on the type.

```graphql
type FileInfo
```
In subgraphs:
```graphql
type FileInfo @key(fields: "fileKey") {
    fileKey: String!
}
```
```graphql
type FileInfo @key(fields: "fileKey", resolvable: false) {
    fileKey: String!
}
```
```graphql
type FileInfo @key(fields: "fileKey", resolvable: false) @shareable {
    fileKey: String!
}
```
```graphql
type FileInfo @key(fields: "fileKey", resolvable: false) @shareable {
    fileKey: String!
    a: String!
    b: String!
}
```
```graphql
type FileInfo @shareable {
    fileKey: String!
    a: String!
    b: String!
}
```

I get an error when trying to execute a query working in apollo loading this type where fields from another subgraph need to be concatenated (regardless of nesting).
"message":"failed to plan operation","error":"upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]"
"message":"internal error", "error":"1 error occurred:\n\t* upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]\n\n"

Unfortunately I couldn't figure out what the exact cause is, sometimes it works, sometimes it doesn't, when rebuilding the schema.
Maybe you'll have an idea what happened. Apollo's working.

## Expected Result

work query

## Actual Result

"message":"failed to plan operation","error":"upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]"
"message":"internal error", "error":"1 error occurred:\n\t* upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]\n\n"

## Environment
OS: (e.g., "Ubuntu 20.04") any
Package Manager: npm
Compiler(if manually compiled): (e.g., "go 14.2") go1.22.3

### Router configuration

```yaml
any
```

### Router execution config

```json
any
```

### Log output

```shell
"message":"failed to plan operation","error":"upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]"
"message":"internal error", "error":"1 error occurred:\n\t* upstream operation: validation failed: external: inline fragment on type: FileInfo mismatches enclosing type: _Entity, locations: [], path: [query,_entities]\n\n"
```

### Additional context

-

**Comment by StarpTech** [comment]
@flymedllva could you please share your schemas with us? You can get it by running `npx wgc@latest federated-graph fetch <name>`

**Comment by flymedllva** [comment]
Unfortunately, I can't. Provided the type with which there are problems with any type of loading.

**Comment by flymedllva** [comment]
After we set the key and removed the sharable the problem went away.

**Comment by Aenimus** [comment]
Hi @flymedllva 

If you cannot share the exact schema/configuration, please attach a minimal replication complete with configuration.

**Comment by flymedllva** [comment]
I wrote in the issue that I couldn't reproduce it separately, but I thought I'd better let you know that the problem exists.

**Comment by Aenimus** [comment]
I've managed to create a minimal replication. It's not the same error as yours, but I believe the root cause is the same.

**Comment by Aenimus** [comment]
I believe this is resolved with router `[0.90.2](https://github.com/wundergraph/cosmo/releases/tag/router%400.90.2)`.

**Comment by flymedllva** [comment]
Why don't we close it then? If I have a similar problem again, I'll create a new one.

Thank you!

**Comment by Aenimus** [comment]
@flymedllva I was hoping you could confirm this is fixed by checking with 0.90.2 before I close.

**Comment by flymedllva** [comment]
Had to quickly finalize the schematic to a working one after [this comment](https://github.com/wundergraph/cosmo/issues/824#issuecomment-2135995315)
Even before editing it did not always work, after reassembling the circuit sometimes it worked, sometimes it did not.
So I can't check it unfortunately(
